test/e2e/e2e_test.go

```diff
@@ -1225,42 +1231,49 @@ func (t *PorchSuite) TestDeleteFromMain(ctx context.Context) {
 
 	t.mustExist(ctx, client.ObjectKey{Namespace: t.namespace, Name: createdSecond.Name}, &pkgSecond)
 
-	// We need to wait for the sync for the "main" revisions to get created
-	time.Sleep(75 * time.Second)
-
-	var list porchapi.PackageRevisionList
-	t.ListE(ctx, &list, client.InNamespace(t.namespace))
-
+	t.Log("Wait for the 'main' revisions to get created")
 	var firstPkgRevFromMain porchapi.PackageRevision
 	var secondPkgRevFromMain porchapi.PackageRevision
-
-	for _, pkgrev := range list.Items {
-		if pkgrev.Spec.PackageName == packageNameFirst && pkgrev.Spec.Revision == "main" {
-			firstPkgRevFromMain = pkgrev
+	var list porchapi.PackageRevisionList
+	finalDeadline := time.Now().Add(5 * time.Minute)
```

Comment:
Could this be fulfilled by the helper function waitUntilMainBranchPackageRevisionExists?

Reply:
I haven't looked into it yet, but just judging by the fact that you proposing it and by the name of the function I am pretty sure it can :)
Thanks, I will fix it.